eBPF.md

```diff
@@ -0,0 +1,179 @@
+
+## ALU Instructions
+
+### 64-bit
+
+Opcode | Mnemonic      | Pseudocode
+-------|---------------|-----------------------
+0x07   | add dst, imm  | dst += imm
+0x0f   | add dst, src  | dst += src
+0x17   | sub dst, imm  | dst -= imm
+0x1f   | sub dst, src  | dst -= src
+0x27   | mul dst, imm  | dst *= imm
+0x2f   | mul dst, src  | dst *= src
+0x37   | div dst, imm  | dst /= imm
+0x3f   | div dst, src  | dst /= src
+0x47   | or dst, imm   | dst \|= imm
+0x4f   | or dst, src   | dst \|= src
+0x57   | and dst, imm  | dst &= imm
+0x5f   | and dst, src  | dst &= src
+0x67   | lsh dst, imm  | dst <<= imm
+0x6f   | lsh dst, src  | dst <<= src
+0x77   | rsh dst, imm  | dst >>= imm (logical)
+0x7f   | rsh dst, src  | dst >>= src (logical)
+0x87   | neg dst       | dst = -dst
+0x9f   | mod dst, imm  | dst %= imm
+0x97   | mod dst, src  | dst %= src
+0xaf   | xor dst, imm  | dst ^= imm
+0xa7   | xor dst, src  | dst ^= src
+0xbf   | mov dst, imm  | dst = imm
+0xb7   | mov dst, src  | dst = src
+0xcf   | arsh dst, imm | dst >>= imm (arithmetic)
+0xc7   | arsh dst, src | dst >>= src (arithmetic)
+
+### 32-bit
+
+These instructions use only the lower 32 bits of their operands and zero the
+upper 32 bits of the destination register.
+
+Opcode | Mnemonic        | Pseudocode
+-------|-----------------|------------------------------
+0x04   | add32 dst, imm  | dst += imm
+0x0c   | add32 dst, src  | dst += src
+0x14   | sub32 dst, imm  | dst -= imm
+0x1c   | sub32 dst, src  | dst -= src
+0x24   | mul32 dst, imm  | dst *= imm
+0x2c   | mul32 dst, src  | dst *= src
+0x34   | div32 dst, imm  | dst /= imm
+0x3c   | div32 dst, src  | dst /= src
+0x44   | or32 dst, imm   | dst \|= imm
+0x4c   | or32 dst, src   | dst \|= src
+0x54   | and32 dst, imm  | dst &= imm
+0x5c   | and32 dst, src  | dst &= src
+0x64   | lsh32 dst, imm  | dst <<= imm
+0x6c   | lsh32 dst, src  | dst <<= src
+0x74   | rsh32 dst, imm  | dst >>= imm (logical)
+0x7c   | rsh32 dst, src  | dst >>= src (logical)
+0x84   | neg32 dst       | dst = -dst
+0x94   | mod32 dst, imm  | dst %= imm
+0x9c   | mod32 dst, src  | dst %= src
+0xa4   | xor32 dst, imm  | dst ^= imm
+0xac   | xor32 dst, src  | dst ^= src
+0xb4   | mov32 dst, imm  | dst = imm
+0xbc   | mov32 dst, src  | dst = src
+0xc4   | arsh32 dst, imm | dst >>= imm (arithmetic)
+0xcc   | arsh32 dst, src | dst >>= src (arithmetic)
+
+### Byteswap instructions
+
+Opcode           | Mnemonic | Pseudocode
+-----------------|----------|-------------------
+0xd4 (imm == 16) | le16 dst | dst = htole16(dst)
+0xd4 (imm == 32) | le32 dst | dst = htole32(dst)
+0xd4 (imm == 64) | le64 dst | dst = htole64(dst)
+0xdc (imm == 16) | be16 dst | dst = htobe16(dst)
+0xdc (imm == 32) | be32 dst | dst = htobe32(dst)
+0xdc (imm == 64) | be64 dst | dst = htobe64(dst)
+
+## Memory Instructions
+
+Opcode | Mnemonic              | Pseudocode
+-------|-----------------------|--------------------------------
+0x18   | lddw dst, imm         | dst = imm
```

Comment:
looks great! probably worth mentioning that this is only 16-byte insn so far. The rest are 8-byte.